### ISSUE TITLE
Poor performance

### ISSUE THREAD

**Issue by clemtrib-leeroy** [born05/craft-assetusage#38]
Same https://github.com/born05/craft-assetusage/issues/32

We will deactivate this plugin because we have timeouts and the asset list doesn't load when the plugin is activate.

Enabling the plugin and switching on the usage columns causes the asset index screens to load very slowly.

Craft CMS 4.4.6 | Asset Usage 3.1.0 | PHP version 8.2.2

**Comment by engram-design** [comment]
Found the same thing, massive timeouts with some folder that make this plugin unusable, sorry to say!

**Comment by MoritzLost** [comment]
Same here. As soon as the usage column is enabled, a folder with <100 files goes from a couple hundred miliseconds to multiple seconds load time.

@roelvanhintum I had a look how the count is implemented, pretty sure this is the culprit:

https://github.com/born05/craft-assetusage/blob/craft4/src/services/Asset.php#L30-L39

The additional call to `Craft::$app->elements->getElementById` means that every single element related to any asset in the list is queried and loaded individually.

The service uses a direct query against the `relations` table. Maybe instead of filtering the results after this query, the function can join the `elements` table to the initial query? This one has the `canonicalId`, `draftId` and `revisionId` fields, should be able to exclude revisions this way, right?

**Comment by ishetnogferre** [comment]
Same here, on sites with a lots of content, it really makes the CMS unusable. We ran into performance issues because of running a LIKE '%__%' with no index.

On Craft 5.

**Comment by MangoMarcus** [comment]
Same, the column feature is unusable for us

**Comment by The-Dan-Main** [comment]
same here, in a folder with 57 Images, this leads to loading times from 1.5-2.5 minutes.

**Comment by kaspar-allenbach** [comment]
yeah mee too. this is a very useful plugin. but it ad heavy strain on the entire system
